play-json/src/main/scala/play/api/libs/json/JsConstraints.scala

```diff
@@ -173,6 +190,9 @@ trait PathWrites {
   def at[A](path: JsPath)(implicit wrs: Writes[A]): OWrites[A] =
     OWrites[A] { a => JsPath.createObj(path -> wrs.writes(a)) }
 
+  def nonDefault[A](path: JsPath, defaultValue: => A)(implicit wrs: Writes[A]): OWrites[A] =
```

Comment:
If am not mistaken this was mentioned earlier. The use cas as I could presume is optimization of traffic between `api client` and `api server`.. I agree that it's may be out of the scope.